### PR TITLE
Fix calculating cart after trying to add more product than in stock

### LINKED ISSUE #14723: TAX problem when we quantity greater than stock in cart
Hi Everyone !

When in the cart, we have a quantity greater than the stock, the subtotal and the total are updated but not the tax.

The problem is on Sylius 1.12 and 1.10, i don'k now for the 1.11.
image
image

(in my example, I have 7 in stock, my subtotal is updated with 8 in quantity, however the tax is not updated)

I think it should update everything or update nothing. 

Thanks ! 
Jérémy.C

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #14723, successor of #14795
| License         | MIT

### CODE REVIEW

File: features/inventory/cart_inventory/verify_inventory_quantity_when_updating_cart_summary.feature [hunk 1]
```diff
@@ -9,13 +9,29 @@ Feature: Verifying inventory quantity on cart summary
...
+        And my cart's total should be "$37.62"
+
+    @ui @api
+    Scenario: Being unable to save quantity that is out of stock and reverting values to the previous ones
+        Given I have added 3 products "Iron Maiden T-Shirt" in the cart
+        And I have added 1 products "Black Dress" in the cart
+        When I change product "Iron Maiden T-Shirt" quantity to 4 in my cart
+        And I change product "Black Dress" quantity to 11 in my cart
+        Then I should be notified that this product cannot be updated
+        And I should see "Iron Maiden T-Shirt" with quantity 3 in my cart
```
Comment (Rafikooo) [inline features/inventory/cart_inventory/verify_inventory_quantity_when_updating_cart_summary.feature:32]: Shouldn't be 4? Shouldn't only revert quantity for "Black Dress"?

Comment (Rafikooo) [inline features/inventory/cart_inventory/verify_inventory_quantity_when_updating_cart_summary.feature:32]: Shouldn't be 4? Shouldn't only revert quantity for the "Black Dress"?

**Review by NoResponseMate — APPROVED** [review]

File: features/inventory/cart_inventory/verify_inventory_quantity_when_updating_cart_summary.feature [hunk 2]
```diff
@@ -9,13 +9,27 @@ Feature: Verifying inventory quantity on cart summary
...
         Given I have added 3 products "Iron Maiden T-Shirt" in the cart
         When I change product "Iron Maiden T-Shirt" quantity to 6 in my cart
         And I update my cart
-        Then I should be notified that this product cannot be updated
+        Then I should be notified that this product has insufficient stock
+
+    @ui @no-api
+    Scenario: Preventing the cart recalculation when the form has errors
+        Given I have added 3 products "Iron Maiden T-Shirt" in the cart
+        And I have added 1 products "Black Dress" in the cart
```
Comment (GSadee) [inline features/inventory/cart_inventory/verify_inventory_quantity_when_updating_cart_summary.feature:26]:
```suggestion
        And I have added a single product "Black Dress" in the cart
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jacob! :1st_place_medal:

**Pull request merged**